src/lib/module-info.ts

```diff
@@ -123,15 +116,24 @@ function properModuleName(folderName: string, fileName: string): string {
     return part === "." ? folderName : joinPaths(folderName, part);
 }
 
-/** Given "foo/bar/baz", return "foo". */
-function rootName(importText: string): string {
+/**
+ * "foo/bar/baz" -> "foo"; "@foo/bar/baz" -> "@foo/bar"
+ * Note: Throws an error for references like
+ */
+function rootName(importText: string, typeFiles: Map<string, unknown>): string {
     let slash = importText.indexOf("/");
     // Root of `@foo/bar/baz` is `@foo/bar`
     if (importText.startsWith("@")) {
         // Use second "/"
         slash = importText.indexOf("/", slash + 1);
     }
-    return slash === -1 ? importText : importText.slice(0, slash);
+    const root = importText.slice(0, slash);
+    const postImport = importText.slice(slash + 1);
+    if (slash > -1 && postImport.match(/v\d+$/) && !typeFiles.has(postImport + ".d.ts")) {
```

Comment:
this error is new, and required a couple of packages on DT to be fixed.

Reply:
Much later: This should be legal if the import is from the same package, eg "foo/v3" should be legal inside "foo".
Not sure how to detect this yet.